pkg/customsignaturestore/customsignaturestore.go

```diff
@@ -3,49 +3,56 @@ package customsignaturestore
 
 import (
 	"context"
+	"crypto/tls"
+	"crypto/x509"
 	"errors"
 	"fmt"
+	"net/http"
 	"net/url"
 	"strings"
 	"sync"
 
+	configv1 "github.com/openshift/api/config/v1"
 	configv1listers "github.com/openshift/client-go/config/listers/config/v1"
 	"github.com/openshift/library-go/pkg/verify/store"
 	"github.com/openshift/library-go/pkg/verify/store/parallel"
 	"github.com/openshift/library-go/pkg/verify/store/sigstore"
+	"golang.org/x/net/http/httpproxy"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/transport"
 )
 
 type Store struct {
 	// Name is the name of the ClusterVersion object that configures this store.
 	Name string
 
-	// Lister allows the store to fetch the current ClusterVersion configuration.
-	Lister configv1listers.ClusterVersionLister
+	// ClusterVersionLister allows the store to fetch the current ClusterVersion configuration.
+	ClusterVersionLister configv1listers.ClusterVersionLister
 
-	// HTTPClient is called once for each Signatures call to ensure
-	// requests are made with the currently-recommended parameters.
-	HTTPClient sigstore.HTTPClient
+	// ConfigMapLister allows the store to fetch certificate
+	// authority ConfigMaps for per-store trust.
+	ConfigMapLister corev1listers.ConfigMapNamespaceLister
+
+	// proxyLister allows the store to fetch the current proxy configuration.
+	proxyLister configv1listers.ProxyLister
```

Comment:
You cannot get Proxy information out of `ClusterVersionLister` or `ConfigMapLister`.  I'd expect us to need this to be a public `ProxyLister` so we could fall back to the Proxy-configured CA set when the user didn't specify a custom `CA`.  Like:

* If the store declares a custom CA set, use that.  Otherwise fall back to...
* If Proxy declares a system-scoped CA set, use that.  Otherwise fall back to...
* The container image's default CA set.

Regardless of CAs, we'll also still need to consume Proxy to set up the HTTP(S) proxy portions.  That's happening today inside the:

```go
httpClientConstructor := sigstore.NewCachedHTTPClientConstructor(optr.HTTPClient, nil)
```

in `cvo.go`, so you might want to keep `HTTPClient` in this structure, continue to populate it with `httpClientConstructor`, drop `proxyLister` (because `HTTPClient` will now handle that part for you), and stuff in the custom CA stuff into the client only when the store sets a custom CA.